pintc/src/predicate/optimize/const_folding.rs

```diff
@@ -46,3 +47,218 @@ pub(crate) fn fold_consts(contract: &mut Contract) {
         }
     }
 }
+
+/// In a given contract, simplify any binary ops that consist of boolean or arithmetic (real and int) identities.
+///
+/// Ex:
+/// true || <expr> is true
+/// true && <expr> is <expr>
+/// false || <expr> is <expr>
+/// false && <expr> is <false>
+/// x + 0 is x
+/// x - 0 is x
+/// x * 0 is 0
+/// x * 1 is x
+/// 0 / x is 0
+/// x / 1 is x
+/// x % 1 is x
+///
+/// Performs a depth-first search to find all folding opportunities.
```

Comment:
Is this an appropriate way to document the identities we handle? Since we check for a specific conditions/values, I feel that it has to be commented somewhere. I'm not sure if it belongs in the function documentation or inline where we handle the identity in the function itself.

If we don't mark them, I can see it being a pain in the rear trying to add new cases in the future. Someone would have to figure out what is already there and then get to work adding more.

Reply:
Yeah this is fine :)